src/compiler/crystal/codegen/link.cr

```diff
@@ -209,7 +209,11 @@ module Crystal
       types.try &.each_value do |type|
         next if type.is_a?(AliasType) || type.is_a?(TypeDefType)
 
-        if type.is_a?(LibType) && type.used? && (link_annotations = type.link_annotations)
+        if type.is_a?(LibType) &&
+           # In interpreted mode we want to load a library even if it's not used,
+           # because it might be used later on, interactively.
+           (interpreted? || type.used?) &&
```

Comment:
I don't think this is a good idea. Interpreted programs should behave as much as compiled ones as possible.
When loading all libraries unconditionally the loader will error on any missing lib even if it's not even used. In a compiled program libraries are only linked when they're actually used. So you might get a program that runs fine in compiled mode, but fails in interpreted mode (for no reason).

Reply:
Okay, I'll undo this change.